File: backend/src/main/java/com/georeferencias/service/AuthService.java
package com.georeferencias.service;

import com.georeferencias.dto.*;
import com.georeferencias.entity.Usuario;

public interface AuthService {
    LoginResponse login(LoginRequest request);
    LoginResponse refreshToken(RefreshTokenRequest request);
    void logout(String token);
    void cambioContrasena(String username, CambioContrasenaRequest request);
    void recuperarContrasena(String email);
    Usuario getUsuarioActual(String username);
}

File: mobile/lib/features/auth/data/auth_service.dart
import 'package:dio/dio.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../../../core/network/dio_client.dart';
import 'auth_models.dart';

final authServiceProvider = Provider<AuthService>((ref) {
  return AuthService(dio: ref.watch(dioProvider));
});

class AuthService {
  final Dio dio;
  AuthService({required this.dio});

  Future<LoginResponse> login(String username, String password) async {
    final response = await dio.post('/auth/login', data: LoginRequest(username: username, password: password).toJson());
    return LoginResponse.fromJson(response.data);
  }

  Future<LoginResponse> refreshToken(String refreshToken) async {
    final response = await dio.post('/auth/refresh', data: RefreshTokenRequest(refreshToken: refreshToken).toJson());
    return LoginResponse.fromJson(response.data);
  }

  Future<UsuarioDTO> getMe() async {
    final response = await dio.get('/auth/me');
    return UsuarioDTO.fromJson(response.data);
  }

  Future<void> logout() async {
    try {
      await dio.post('/auth/logout');
    } catch (_) {}
  }
}
